### PR TITLE
Agenda date list and actions added

### PR DESCRIPTION
**Pull request by rokal**
agenda1_view

### CODE REVIEW

File: app/src/main/java/com/roosongo/tasky/agenda/presentation/Agenda.kt [hunk 1]
```diff
@@ -1,70 +1,75 @@
...
+    AgendaToolbar(currentDate = viewModel.agendaState.currentDate) {
+      viewModel.selectNewDate(it)
+    }
+    Container {
+      AgendaDates(
+        dates = viewModel.agendaState.agendaDates,
+        currentDate = viewModel.agendaState.currentDate
+      ) {
+        viewModel.updateCurrentDate(it)
+      }
```
Comment (philipplackner) [inline app/src/main/java/com/roosongo/tasky/agenda/presentation/Agenda.kt:50]: Same comment as in the last PR regarding named parameters 👍

File: app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.time.LocalDate
+import javax.inject.Inject
+
+fun generateNextXDaysFrom(date: LocalDate, x: Int): ImmutableList<LocalDate> {
+  var result: MutableList<LocalDate> = mutableListOf()
+  for (i in 1..x) {
+    result.add(date.plusDays((i - 1).toLong()))
+  }
+  return result.toImmutableList()
+}
```
Comment (philipplackner) [inline app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt:19]: This is rather business logic and should go in domain. Also, I'm not a big fan of using mutable lists (even though you did it 100% correct here with the conversion). I like using the list functions the Kotlin standard lib offers us, like this for example which is quite shorter:
```
return (1..x).map { date.plusDays((it - 1).toLong) }
```

File: app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      currentDate = newDate
+    )
+  }
+
+  fun selectNewDate(newDate: LocalDate) {
+    agendaState = agendaState.copy(
+      currentDate = newDate,
+      agendaDates = generateNextXDaysFrom(newDate, 6)
+    )
+  }
```
Comment (philipplackner) [inline app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt:43]: I'd also allow the user to go back some days. Currently, it seems if they select the day 6 days after today, that one will be the first start date (and therefore be on the left) leaving them no easy chance of switching back.
I personally didn't generate any future days at all when the user clicks on one explicitly. I only do that when they select a day from the day picker dialog.

But this is not clear from the requirements, so this is totally up to you and how you like it :)

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt (hunk 3 above)
Comment (rokal) [reply app/src/main/java/com/roosongo/tasky/agenda/presentation/AgendaViewModel.kt:43]: this https://github.com/rokal/Tasky/pull/4/files#diff-f2c88dacbdfd09827911360bdb631fa6fe858ec61161bb417d4f859e7c6eefb7R32 function is the one used when the user clicks on the a date on the calendar view. If the user clicks on the month dropdown and picks a new date, then new dates are generated using the `https://github.com/rokal/Tasky/pull/4/files#diff-f2c88dacbdfd09827911360bdb631fa6fe858ec61161bb417d4f859e7c6eefb7R38`

Any way the process can be improved even if the user selects a new date that is already in the visible calendar range.

**Pull request merged**